File: src/dataset/typescript/space-age_Exercism-typescript-FilipeCerejo/claude/ClaudeSonnet4_space-age_v4.ts
const OrbitalPeriods = {
  'mercury': 0.2408467,
  'venus': 0.61519726,
  'earth': 1,
  'mars': 1.8808158,
  'jupiter': 11.862615,
  'saturn': 29.447498,
  'uranus': 84.016846,
  'neptune': 164.79132
} as const;

const EARTH_YEAR_SECONDS = 31557600;
const SCALE_FACTOR = 100;

const precomputedFactors = {
  'mercury': SCALE_FACTOR / (OrbitalPeriods.mercury * EARTH_YEAR_SECONDS),
  'venus': SCALE_FACTOR / (OrbitalPeriods.venus * EARTH_YEAR_SECONDS),
  'earth': SCALE_FACTOR / (OrbitalPeriods.earth * EARTH_YEAR_SECONDS),
  'mars': SCALE_FACTOR / (OrbitalPeriods.mars * EARTH_YEAR_SECONDS),
  'jupiter': SCALE_FACTOR / (OrbitalPeriods.jupiter * EARTH_YEAR_SECONDS),
  'saturn': SCALE_FACTOR / (OrbitalPeriods.saturn * EARTH_YEAR_SECONDS),
  'uranus': SCALE_FACTOR / (OrbitalPeriods.uranus * EARTH_YEAR_SECONDS),
  'neptune': SCALE_FACTOR / (OrbitalPeriods.neptune * EARTH_YEAR_SECONDS)
} as const;

type OrbitalPeriodType = keyof typeof OrbitalPeriods;

export function age(planet: string, seconds: number): number {
    const factor = precomputedFactors[planet as OrbitalPeriodType];
    return Math.round(seconds * factor) / SCALE_FACTOR;
}

File: src/dataset/go/space_age_Exercism_go_rootulp/claude/ClaudeSonnet4_space_age_v4.go
package space

type Planet string

const (
	secondsInEarthYear = 31557600.0
	
	Mercury Planet = "Mercury"
	Venus   Planet = "Venus"
	Earth   Planet = "Earth"
	Mars    Planet = "Mars"
	Jupiter Planet = "Jupiter"
	Saturn  Planet = "Saturn"
	Uranus  Planet = "Uranus"
	Neptune Planet = "Neptune"
)

var orbitalPeriods = map[Planet]float64{
	Mercury: 0.2408467,
	Venus:   0.61519726,
	Earth:   1.0,
	Mars:    1.8808158,
	Jupiter: 11.862615,
	Saturn:  29.447498,
	Uranus:  84.016846,
	Neptune: 164.79132,
}

func Age(ageInSeconds float64, planet Planet) float64 {
	return (ageInSeconds / secondsInEarthYear) / orbitalPeriods[planet]
}

func ageOnEarth(ageInSeconds float64) float64 {
	return ageInSeconds / secondsInEarthYear
}